### PR TITLE
Add an LRUMap implementation based on cash/lrucache

### PR DESCRIPTION
**Pull request by treyhyde**
LRUMap for discussion

---

This change is [Reviewable](https://reviewable.io/reviews/schmittjoh/php-collection/17)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -9,9 +9,13 @@
     "require": {
         "phpoption/phpoption": "1.*"
     },
+    "require-dev": {
+        "phpunit/phpunit": "3.*"
+    },
     "autoload": {
         "psr-0": {
-            "PhpCollection": "src/"
+            "PhpCollection": ["src/", "tests/"]
```
Comment (stof) [inline composer.json]: this is wrong. Tests should be autoloaded in autoload-dev, not in autoload

File: composer.json [hunk 2]
```diff
@@ -9,9 +9,13 @@
     "require": {
         "phpoption/phpoption": "1.*"
     },
+    "require-dev": {
+        "phpunit/phpunit": "3.*"
```
Comment (stof) [inline composer.json]: phpunit 3 is years old. The stable version is 4.4 currently

File: src/PhpCollection/LRUMap.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        $this->maximumSize = $size;
+
+        $this->elements = array();
+    }
+
+    /**
+     * Get the value cached with this key
+     *
+     * @param int|string $key     The key. Strings that are ints are cast to ints.
+     * @param mixed      $default The value to be returned if key not found. (Optional)
```
Comment (stof) [inline src/PhpCollection/LRUMap.php]: this does not exist

File: tests/PhpCollection/Tests/LRUMapTest.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace PhpCollection\Tests;
+
+use PhpCollection\LRUMap;
+use PhpCollection\Map;
+use PhpCollection\Tests\MapTest;
+
+class LRUMapTest extends MapTest
```
Comment (stof) [inline tests/PhpCollection/Tests/LRUMapTest.php]: why extending from MapTest ? This will run the MapTest twice, which does not make sense

Comment (stof) [inline tests/PhpCollection/Tests/LRUMapTest.php]: ah sorry, I missed the overwrite of setUp changing the map implementation

File: tests/PhpCollection/Tests/MapTest.php [hunk 5]
```diff
@@ -7,7 +7,7 @@
 class MapTest extends \PHPUnit_Framework_TestCase
 {
     /** @var Map */
-    private $map;
+    public $map;
```
Comment (stof) [inline tests/PhpCollection/Tests/MapTest.php]: you should keep it protected rather than public

File: composer.json (hunk 2 above)
Comment (treyhyde) [inline composer.json]: yeah, sorry,  habit, phpunit dropped too much functionality for us to use 4.x.  Will fix, should be fine here.

**Comment by treyhyde** [comment]
I think I covered all your issues here, thanks for the feedback.

**Pull request closed without merging**